File: Components/DevExpressVCL/Demos/ExpressBars/CBuilder/InPlaceEditorsDemo/InPlaceEditorsDemoValue.cpp
//---------------------------------------------------------------------------

#include <vcl.h>
#pragma hdrstop

#include "InPlaceEditorsDemoValue.h"
//---------------------------------------------------------------------------
#pragma package(smart_init)
#pragma resource "*.dfm"
TfrmValueEditors *frmValueEditors;
//---------------------------------------------------------------------------
__fastcall TfrmValueEditors::TfrmValueEditors(TComponent* Owner)
	: TEditorDemoBaseFrame(Owner)
{
}
//---------------------------------------------------------------------------
void __fastcall TfrmValueEditors::PaintBox1Paint(TObject *Sender)
{
  PaintBox1->Canvas->Font->Name = FFontName;
  PaintBox1->Canvas->Font->Size = FFontSize;
  AnsiString AStrings[3] = {FDate, FTime, FValue};
  DrawText(AStrings, PaintBox1, 3);
}
//---------------------------------------------------------------------------
void __fastcall TfrmValueEditors::SetParameters(int AFontSize, AnsiString ADate, AnsiString ATime, Currency AMoney)
{
  FFontName = "Times New Roman";
  FFontSize = (AFontSize + 8) / 2;
  FDate = "Date: " + ADate;
  FTime = "Time: " + ATime;
  TVarRec args = AMoney;
  FValue = Format("Value: %m", &args, 0);
  PaintBox1->Invalidate();
}

File: Components/DevExpressVCL/Demos/ExpressBars/Delphi/InPlaceEditorsDemo/InPlaceEditorsDemoValue.pas
unit InPlaceEditorsDemoValue;

interface

uses
  Windows, Messages, SysUtils, Classes, Graphics, Controls, Forms,
  Dialogs, StdCtrls, cxGeometry, ExtCtrls, ComCtrls,
  InPlaceEditorsDemoFrameManager;

type
  TfrmValueEditors = class(TEditorDemoBaseFrame)
    PaintBox1: TPaintBox;
    procedure PaintBox1Paint(Sender: TObject);
  private
    FDate, FTime, FValue: string;
    FFontName: string;
    FFontSize: Integer;
  public
    procedure SetParameters(AFontSize: Integer; ADate, ATime: string; AMoney: Currency);
  end;

implementation

{$R *.dfm}

{ TfrmValueEdits }

procedure TfrmValueEditors.SetParameters(AFontSize: Integer; ADate, ATime: string; AMoney: Currency);
begin
  FFontName := 'Times New Roman';
  FFontSize := (AFontSize + 8) div 2;
  FDate := Format('Date: %s', [ADate]);
  FTime := Format('Time: %s', [ATime]);
  FValue := Format('Value: %m', [AMoney]);

  PaintBox1.Invalidate;
end;

procedure TfrmValueEditors.PaintBox1Paint(Sender: TObject);
begin
  PaintBox1.Canvas.Font.Name := FFontName;
  PaintBox1.Canvas.Font.Size := FFontSize;
  DrawText([FDate, FTime, FValue], PaintBox1);
end;

end.
